### ISSUE TITLE
Table in mysql DB not copied from source to destination 

### ISSUE THREAD

**Issue by netzzer** [fusor/mig-controller#240]
Using gluster-mysql-persistent.yaml template created mysql deployment on OCP3 (oc new-project mysql-rwo-annotate; oc create -f gluster-mysql-persistent.yaml). After deployment was done logged into mysql pod (oc rsh <mysql_pod_name; mysql -u root) and create a new table with data.

```
mysql> use sampledb;
Database changed
mysql> show tables;
Empty set (0.01 sec)
mysql> create table test (col1 VARCHAR(20), col2 VARCHAR(20));
Query OK, 0 rows affected (0.10 sec)
mysql> insert into test values ('data', 'moredata');
Query OK, 1 row affected (0.02 sec)
mysql> select * from test;
+------+----------+
| col1 | col2     |
+------+----------+
| data | moredata |
+------+----------+
1 row in set (0.01 sec)
```

Then successfully Stage+Migrate using CAM from OCP3 to OCP4. After migration checked to see if new table was in mysql sampledb. The table did not exist. See graphic. 

![mysql-add-table-ocp3-ocp4](https://user-images.githubusercontent.com/5932489/62487142-f0c4d980-b775-11e9-8ab5-9a1b1a6ba329.jpg)

[gluster-mysql-persistent.yaml.gz](https://github.com/fusor/mig-controller/files/3468891/gluster-mysql-persistent.yaml.gz)

**Comment by netzzer** [comment]
Also tried with same gluster-mysql-persistent.yaml and change accessMode: ReadWriteMany and created on OCP3 cluster. Added two tables (test and test1) with data in each table to mysql sampledb on OCP3 side. Did successful Stage+Migrate to OCP4 and mysql cephfs volume. Logged into new mysql pod. Same results as using ReadWriteOnce accessMode and cephrbd volume, there were no tables in mysql sampledb. 

Before starting Stage I redirected the logs for each of the velero pods on OCP3 and OCP4 to a file. Logs are attached. 
[velero-logs.tar.gz](https://github.com/fusor/mig-controller/files/3469053/velero-logs.tar.gz)

**Comment by dymurray** [comment]
I finally got an environment to test this (gluster->ceph) and am seeing the same thing with an MSSQL application. I am debugging it and will update the issue when I know what is causing it.

**Comment by dymurray** [comment]
Erik confirmed for me on `latest` he was able to add tables to MySQL and migrate as expected going from NFS->EBS which tells me this is a Gluster specific problem. My first guess is that the restic nodes are not updating their mounts properly to the Gluster volume. The reason I'm thinking it's that direction is this is exactly what mount propagation errors were causing with NFS for me. However, bouncing the restic pods didn't seem to solve this problem so I'm going to run a few more tests and update the issue with more information.

**Comment by sseago** [comment]
In this morning's testing I realized that my nfs src was failing in the same way as the glusterfs and glusterblock sources (all migrating to cephrbd or cephfs targets). I re-ran a migration selecting gp2/ebs as my destination storage class and everything worked. Table data was properly copied for all 3 source types: nfs, glusterfs, and glusterblock. It looks like our bug is with cephfs and rbd volumes, not gluster ones.

**Comment by dymurray** [comment]
Updating this issue with what we have learned. This bug is due to Velero not restoring data properly to CSI mounted volumes. We are working with the upstream Velero team to get this resolved. Will either resolve this in our fork or upstream.

**Comment by sseago** [comment]
Velero upstream has fixed the CSI problem for 1.1, but there is one outstanding bug in the 1.1 codebase preventing us from adopting it. The new functionality has introduced a bug that causes volume restores to fail if Velero is running in a non-standard namespace (such as the 'mig' namespace where the migration operator installs it). Upstream has reproduced the issue and we're working to identify and fix the cause both on our team and upstream.

**Comment by sseago** [comment]
The namespace bug has an upstream PR with a fix that we have tested with success with the mig operator. Once this PR is merged we can get all of this working in our velero fork. There may be another refactoring of the fix before merge, and we'll need to test that one as well.

**Comment by jwmatthews** [comment]
@sseago should we close this issue, or is there still some work we are tracking yet to be done?

**Comment by sseago** [comment]
@jwmatthews This should be done now. The referenced additional upstream bug was included in the Velero 1.1 final release, which we've rebased to on our fork.
